conn_pool.go

```diff
@@ -335,8 +398,16 @@ func (p *ConnPool) Prepare(name, sql string) (*PreparedStatement, error) {
 	if err != nil {
 		return nil, err
 	}
-	ps, err := c.Prepare(name, sql)
 	p.availableConnections = append(p.availableConnections, c)
+
+	// Double check that the statement was not prepared by someone else
+	// while we were acquiring the connection (since acquire is not fully
+	// blocking now, see createConnectionUnlocked())
+	if ps, ok := p.preparedStatements[name]; ok && ps.SQL == sql {
+		return ps, nil
+	}
+
+	ps, err := c.Prepare(name, sql)
```

Comment:
Since we already added `c` to `availableConnections`, wont we be running `Prepare` twice on `c`?